hydromt_wflow/components/staticmaps.py

```diff
@@ -1,11 +1,254 @@
+        elif not isinstance(data, xr.Dataset):
+            raise ValueError(f"Cannot set data of type {type(data).__name__}")
+
+        # Check for cyclic data
+        if "time" in data.dims:
+            # Raise error if the dimension does not have a supported length
+            if len(data.time) not in [12, 365, 366]:
+                raise ValueError(
+                    f"Length of cyclic dataset ({len(data.time)}) is not supported by "
+                    "Wflow.jl. Ensure the data has length 12, 365, or 366"
+                )
+            tname = "time"
+            time_axes = {
+                k: v for k, v in self.data.sizes.items() if k.startswith("time")
+            }
+            if data["time"].size not in time_axes.values():
+                tname = f"time_{data['time'].size}" if "time" in time_axes else tname
+            else:
+                k = list(
+                    filter(lambda x: time_axes[x] == data["time"].size, time_axes)
+                )[0]
+                tname = k
+
+            if tname != "time":
+                data = data.rename_dims({"time": tname})
+        if "layer" in data.dims and "layer" in self.data:
+            if len(data["layer"]) != len(self.data["layer"]):
+                vars_to_drop = [
+                    var for var in self.data.data_vars if "layer" in self.data[var].dims
+                ]
+                # Drop variables
+                logger.warning(
+                    f"Replacing 'layer' coordinate, dropping variables \
+({vars_to_drop}) associated with old coordinate"
+                )
+                # Use `_grid` as `grid` cannot be set
+                self._data = self.data.drop_vars(vars_to_drop + ["layer"])
 
-__all__ = ["StaticmapsComponent"]
+        # Determine the masking layer
+        mask = get_mask_layer(mask, self.data, data)
 
+        # Set the data per layer
+        for dvar in data.data_vars:
+            # TODO removed reading mode check, review the exact effect of this
+            # I personally couldn't see the point..
+            if dvar in self._data:
+                logger.warning(f"Replacing grid map: {dvar}")
+            if mask is not None:
+                # TODO maybe we could outlaw boolean layers
+                # Instead use 0 and 1 for actual data and -1 or -9999 for all i care
+                # for the nodata value
+                if data[dvar].dtype != "bool":
+                    data[dvar] = data[dvar].where(mask, data[dvar].raster.nodata)
+                else:
+                    data[dvar] = data[dvar].where(mask, False)
+            self._data[dvar] = data[dvar]
 
-class StaticmapsComponent(GridComponent):
-    """wflow staticmaps component."""
+    ## Setup and update methods
+    @hydromt_step
+    def update_names(
+        self,
+        **select,
+    ):
+        """Map the names of the data variables to new ones.
 
-    pass
+        Parameters
+        ----------
+        select : dict, optional
```

Comment:
It's internal but in the docstrings still so I harmonised with names in other core methods

```suggestion
        rename : dict, optional
```

Reply:
I see you did it below but forgot it in the docstrings :)